shared/message/Geometry.proto

```diff
@@ -24,41 +24,107 @@ package message;
 import "Vector.proto";
 import "Matrix.proto";
 
+/**
+ * A normal vector and the distance to the line.
+ */
 message Line {
     fvec3 normal   = 1;
     float distance = 2;
 }
 
+/**
+ * The circle's centre position and radius.
+ */
 message Circle {
     double radius = 1;
     vec2   centre = 2;
 }
 
+/**
+ * A 3D matrix representation of the Ellipse.
+ */
 message Ellipse {
     // See http://en.wikipedia.org/wiki/Matrix_representation_of_conic_sections
     mat3 ellipse = 1;
 }
 
+/**
+ * Four corner points of a quadrilateral.
+ */
 message Quad {
+
+    /**
+     * Top left corner of the quadrilateral.
+     */
     vec2 tl = 1;
+
+    /**
+     * Top right corner of the quadrilateral.
+     */
     vec2 tr = 2;
+
+    /**
+     * Bottom left corner of the quadrilateral.
+     */
     vec2 bl = 3;
+
+    /**
+     * Bottom right corner of the quadrilateral.
+     */
     vec2 br = 4;
 }
 
+/**
+ * Contains the polygon's location.
+ */
 message Polygon {
+
+    /**
+     * {x, y} location of the polygon.
+     */
     vec2 point = 1;
 }
 
+/**
+ * Contains the positions of the four corners of a frustrum.
+ */
 message Frustum {
+
+    /**
+     * Top left corner of the frustrum
+     */
     vec3 tl = 1;
+
+    /**
+     * Top right corner of the frustrum
+     */
     vec3 tr = 2;
+
+    /**
+     * Bottom left corner of the frustrum
+     */
     vec3 bl = 3;
+
+    /**
+     * Bottom right corner of the frustrum
+     */
     vec3 br = 4;
 }
 
+/**
+ * Contains the cone's axis, gradient, and radius.
+ */
 message Cone {
-    fvec3 axis     = 1;
+
+    /**
+     * 3D axis of the centre of the cone.
+     */
+    fvec3 axis = 1;
+
     float gradient = 2;
-    float radius   = 3;
+
+    /**
+     * Radius of the circle which is the base of the cone.
+     */
+    float radius = 3;
```

Comment:
These really need to be fixed up at some point, gradient and radius are the same thing, but we only use one of them now and having two fields to describe the same thing results in confusion

Reply:
Is that beyond the scope of this PR? I can make an issue